activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -145,5 +150,9 @@ def can_be_bound?(column_name, value)
         handler_for(value).is_a?(BasicObjectHandler) &&
         !table.associated_with?(column_name)
     end
+
+    def polymorphic_array?(value)
+      value.is_a?(Array) && value.map(&:class).map { |e| e.try(:base_class) }.uniq.size > 1
```

Comment:
Thanks @sgrif!  I've addressed your comments in the latest commit.  Mind giving it a look over, would value your feedback on it.  

CI is red for unrelated changes (looks like it's red on the latest run of other PRs for the same timezone related tests).  I can't rebuild, so I'll push an empty commit later, once I see some other builds go green.